### ISSUE TITLE
「YukariWhisper」最高です、ただ一つどうしても搭載してほしいことが・・・

### ISSUE THREAD

**Issue by GeekAssa** [tyapa0/YukariWhisper#7]
「ngwords.txt」とは別の機能というか辞書を搭載することは厳しいでしょうか？
もし可能なら「ngwords.txt」みたいな「Jisho.txt」等のようにTXT形式が望ましいのですが技術的に不可能！？

ゲーム実況などで専門用語等を登録して漢字などを強制的に表示できれば最高なのですが。

**Comment by tyapa0** [comment]
辞書機能はゆかコネNEOやゆかりネットのプラグインで用意されてますので、まずはそちらを試してみてください。
![image](https://github.com/user-attachments/assets/5f075f85-e211-40a1-8215-e819ea5bd25e)

**Comment by GeekAssa** [comment]
なるほどプラグイン側で連携できるとは知りませんでした確認不足すみません。
試してみます、ありがとうございます！！

**Comment by GeekAssa** [comment]
![muridesu](https://github.com/user-attachments/assets/f986d2af-6723-4002-aa95-f73c6d1c4565)
テストしてみましたが結果は駄目でした

使用したものは「ゆかコネNEO(最新)」+ YukariWhisper-0.1.1 + 辞書プラグイン v1.9c
「ゆかコネNEO」、「辞書プラグイン」と「YukariWhisper-0.1.1」これらの連携は出来ているのでしょうか？

どの様になったか画像を添付しておきます。
今回のテストでほしかった結果は

「てすと」と発音すると「0123456789」の結果(字幕)
「ばなな」と発音すると「ばーーなんなん」の結果(字幕)

しかし辞書登録後も「ゆかりネットコネクター」側のウィンドウでも
「てすと」は「テスト」
「ばなな」は「バナナ」
のままでした、何か手順間違えてますでしょうか？

辞書は登録後「SAVE」して「ゆかコネNEO」も「YukariWhisper-0.1.1」も再起動しています。

コマンドプロンプトに出ているように私の声は正常に認識しております。

**Comment by GeekAssa** [comment]
変換されない原因が判明しました「YukariWhisper-0.1.1」のコマンドラインに出力される文字
今回の場合ですと「てすと」と発音したとき　※コマンドライン「YukariWhisper」には「テスト」と出ており
そちら(カタカナ)で辞書登録すると「0123456789」という正常な結果が得られました。

しかし私の求めるものとはかなりかけ離れています。
辞書登録に全て「ひらがな」で登録　→「置き換える単語」今回の例だと
発音「てすと」→「0123456789」にしたいのですが（※これはあくまでも例です）
これら辞書登録プラグインは「YukariWhisper-0.1.1」から渡された「文字列」を変更してくれるだけなのですね。

極端な例を出すと「てすと」と自分が発音して「YukariWhisper-0.1.1」のコマンドライン上では
「テストン」や「テストぉ」や「てすト」と何パターンも発音するたびに違う感じに表示（認識）されてしまった場合
それら全てを「辞書登録プラグイン」に登録しないとだめという状態。

これは私のイメージなのですが日本語を喋ったとき言葉（ひらがな）で認識
その後↓↓↓これらの学習済みのデータにて判断して「ひらがな ＆ カタカナ ＆ 漢字」に置き換えている?と勝手に思ってますが
;whisper_model_size = kotoba-tech/kotoba-whisper-v1.0-faster
whisper_model_size = large-v3
;whisper_model_size = large-v2
;whisper_model_size = medium
;whisper_model_size = small

これらに判断させる前の「ひらがな」だけの状態を「辞書プラグイン」に渡すことはできないでしょうか？

現状のままですと「YukariWhisper-0.1.1」でどのように認識されるか
実際に発音して（コマンドライン上に）表示されたテキストを辞書登録しないと
正常に動いてくれません。
前後の文章で「漢字」が変わる場合、極端な話ですがそれら全て登録しないと思うように動いてくれません。

繰り返しになりますが「ひらがな ＆ カタカナ ＆ 漢字」を学習辞書等で判断する前の
「ひらがな」だけの状態を「辞書プラグイン」へ直接渡すことは仕様上できないのでしょうか？

もしくは、手間かもしれませんが「YukariWhisper-0.1.1」にて「Jisho.txt」等のようにTXT形式で登録で置き換え
「YukariWhisper」内にて全て完結→ゆかコネ、私としてはこれがベストだと素人ながらに思ってしまいます。

**Comment by tyapa0** [comment]
>これは私のイメージなのですが日本語を喋ったとき言葉（ひらがな）で認識

いいえ、残念ながらYukariWhisperは音声データをkotoba-whisperなどのAIに入れて出てきた文字を「そのまま出力」しているだけです。

よって、モデルごとに単語のボキャブラリーや句読点表現が少しずつ異なります。
もしこれを何とかしたいのであればモデルを再学習してチューニングする作業が必要になります。
実際、kotoba-whisperさんは再学習処理をして句読点表現を改善したモデルを出しています。
https://x.com/kotoba_tech/status/1788836511993282804
※計算機を沢山使うので研究機関や企業でないと再学習は難しいです。

>これら辞書登録プラグインは「YukariWhisper-0.1.1」から渡された「文字列」を変更してくれるだけなのですね。

なので、出来たとしても同じように **「渡された「文字列」を変更してくれる」** だけしか出来ません。

YukariWhisperはMIT licenseでコードを公開しております。 forkなどして改善していただければ幸いです。
https://github.com/tyapa0/YukariWhisper/forks

**Comment by GeekAssa** [comment]
>>いいえ、残念ながらYukariWhisperは音声データをkotoba-whisperなどのAIに入れて出てきた文字を「そのまま出力」しているだけです。

なるほど、ただ疑問なのですが「kotoba-whisper」のAI等に入れる**前段階**ではただの「音」つまり
（前者）「あ」なら「あ」、「き」なら「き」なのではないのでしょうか？

（後者）それともAI辞書？は送られてきた音の**要素**（ある程度の**何秒間かのぶつ切り**、もしくは纏まった**単語**等）で判断しているのでしょうか？

もし後者でしたらどうにもなりそうもないですが、AIの学習された辞書で判断する前の音の要素次第で
いけるのではとかってに思えてしまいました。

そんな単純なことではないとは思いますが。
当分は出力される何パターンも登録するしかないようですね。お答え頂きありがとうございました。

**Comment by tyapa0** [comment]
>（後者）それともAI辞書？は送られてきた音の要素（ある程度の何秒間かのぶつ切り、もしくは纏まった単語等）で判断しているのでしょうか？

AIモデルはマイクそのものから判断しています。それを文字列に変換するのが「kotoba-whisper」等のAIモデルの仕事です。
ここはOpenAIなどの企業が開発している部分です。
YukariWhisperはただマイク設定をして、AIから文字列をもらって ゆかコネNEOなどに渡してるだけです。

ソースはここにあるのが全てです、使いやすいよう自由に改変してください。
使いやすいものが出来たら、共有していただければ幸いです。

**Comment by GeekAssa** [comment]
>>AIモデルはマイクそのものから判断しています。それを文字列に変換するのが「kotoba-whisper」等のAIモデルの仕事です。
>>ここはOpenAIなどの企業が開発している部分です。
>>YukariWhisperはただマイク設定をして、AIから文字列をもらって ゆかコネNEOなどに渡してるだけです。

だとするなら
>>ソースはここにあるのが全てです、使いやすいよう自由に改変してください。
>>使いやすいものが出来たら、共有していただければ幸いです。

この言い分はおかしくないでしょうか？我々一般人にはどうすることもできない部分、つまり
日本語（ひらがな）→AIモデルで判断して該当する確率の高い「漢字」や「カタカナ」など判断(変換)して→文字出力。
最後の**AIから出力された文字TXT部分**をこちら**YukariWhisper**で受け取り**ゆかコネNEO**へ渡しているだけですよね？

こちらのソースをどのようにいじって判定部分であるAIパートをいじれるというのでしょうか？
あなたがおっしゃっていることから判断できるのは、それは技術的に不可能だと思います。


現状の仕様を聞く限り「kotoba-whisper」自体が漢字へ**変換する前に認識した音（ひらがな）だけの部分**を
「kotoba-whisper」が仕様変更するなり新たなオプション、例えば平仮名だけを出力するオプション等搭載しないと無理ですよね。
